File: modules/auth/src/main/java/com/media/bus/auth/modules/auth/repository/RoleRepository.java
package com.media.bus.auth.modules.auth.repository;

import com.media.bus.auth.modules.auth.entity.Role;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

/**
 * 역할 마스터 Repository.
 * 회원가입 시 MemberType.name()으로 Role 엔티티를 조회하는 데 사용됩니다.
 */
public interface RoleRepository extends JpaRepository<Role, UUID> {

    Optional<Role> findByName(String name);
}

File: modules/iam/src/main/kotlin/com/media/bus/iam/auth/repository/RoleRepository.kt
package com.media.bus.iam.auth.repository

import com.media.bus.iam.auth.entity.RoleEntity
import com.media.bus.iam.auth.entity.RoleTable
import org.jetbrains.exposed.v1.core.eq
import org.springframework.stereotype.Repository
import java.util.*

/**
 * ## 역할 마스터 Exposed Repository
 *
 * 회원가입 시 `MemberType.name()`으로 Role 엔티티를 조회하는 데 사용한다.
 */
@Repository
class RoleRepository {

    fun findAll(): List<RoleEntity> = RoleEntity.all().toList()

    fun findById(id: UUID): RoleEntity? = RoleEntity.findById(id)

    fun findByName(name: String): RoleEntity? =
        RoleEntity.find { RoleTable.name eq name }.firstOrNull()
}
